### PR TITLE
ACF field updates

### PR DESCRIPTION
**Pull request by jomurgel**
Closes #471 

## Description
Made several general updates to improve the ACF blocks, organization, and add a couple of new features.

### Additions
Replaced the button link/button text with the ACF link field.

screen shot 2019-03-01 at 9 15 09 am

I've then added a new get/display function [_s_get_link](https://github.com/WebDevStudios/wd_s/blob/hotfix/hotfix-030119-acf-field-updates/inc/acf.php#L434) which will output a link from the array and optionally a button class/additional classes.

``` php
// Defaults.
$defaults = array(
	'button' => false, // display as button?
	'class'  => '',
	'link'   => get_sub_field( 'button_link' ),
);
```

Updated the hero, carousel, and CTA blocks that are currently using a button.

Also in use is a [_s_has_array_key()](https://github.com/WebDevStudios/wd_s/blob/hotfix/hotfix-030119-acf-field-updates/inc/template-tags.php#L569) which checks to see if an array key exists, but also if the item has value.

I've also added an "overlay" toggle to the image background which has been a frequent request recently.

screen shot 2019-03-01 at 9 18 25 am

And added that necessary info into the acf.php function: https://github.com/WebDevStudios/wd_s/blob/hotfix/hotfix-030119-acf-field-updates/inc/acf.php#L86

## Updates
Related to this was fixing some issues inside the [_hero.scss](https://github.com/WebDevStudios/wd_s/blob/hotfix/hotfix-030119-acf-field-updates/assets/sass/modules/content-blocks/_hero.scss#L6) part where the actual wrapping container and overlay wasn't, in fact, being rendered because it was expecting `.hero-block` but styling `.hero`. This also addresses a couple of z-indexing issues with the overlay.

I also removed the `$animation_class` variable from ACF template parts and instead inlined the function since they're only called once.

Opened #472 to potentially remove or update animation, but no other changes were made with that.

I removed the "title" field from the generic content block which doesn't exist.

Finally, I renamed the ACF blocks from "Group: Hero — To Clone" to "Block: Hero — To Clone" which organizes the ACF admin into blocks, global, fields, and groups. Easier to find things in my opinion.

One thing I think we COULD do, but I did not, is replace any instance of the title/text combo in blocks (CTA, Hero, Carousel, etc) with a WYSIWYG field to make the general markup and user experience better. Opened an issue to discuss this: #473

### CODE REVIEW

File: inc/acf.php [hunk 1]
```diff
@@ -405,3 +410,69 @@ function _s_get_theme_colors() {
...
+	}
+
+	// Append button class if button exists.
+	$classes = $args['button'] ? ' button' : '';
+
+	// Append class.
+	$classes .= ' ' . $args['class'];
+
+	// Get title else default to "Read More".
+	$title = _s_has_array_key( 'title', $button_array ) ? $button_array['title'] : esc_html__( 'Read More', 'firemon' );
```
Comment (coreymcollins) [inline inc/acf.php]: Need to update this text domain

File: template-parts/content-blocks/block-cta.php [hunk 2]
```diff
@@ -29,8 +26,13 @@
...
 		</header>
 
-		<?php if ( $button_url && $button_text ) : ?>
-			<a class="button cta-button" href="<?php echo esc_url( $button_url ); ?>"><?php echo esc_html( $button_text ); ?></a>
-		<?php endif; ?>
+		<?php
+		_s_display_link(
+			array(
+				'button' => true,
+				'class'  => 'cta-button',
```
Comment (coreymcollins) [inline template-parts/content-blocks/block-cta.php]: Can we change this to `button-cta`? We're using `button-hero` in the Hero file and I'd like to keep them consistent in naming.

File: template-parts/content-blocks/block-carousel.php [hunk 3]
```diff
@@ -24,8 +24,6 @@
 		// Set up fields.
 		$title           = get_sub_field( 'title' );
```
Comment (coreymcollins) [inline template-parts/content-blocks/block-carousel.php]: Need to realign these `=` signs

**Review by coreymcollins — CHANGES_REQUESTED** [review]
Left some code review notes with some additional bits below.

## Overlay Toggle
Can we add this to videos as well? Right now the option only exists for image backgrounds.

## Overlay Display
I have the overlay enabled, but it's not displaying. It looks like the option was added globally, but the styles were only added for the hero.

### Accordion


### CTA


### 50/50


### Generic Content


### Recent Posts


### Related Posts


## Carousel Overlays


I guess we've always had this, since there isn't anything new in your commits, but could be something to clean up as part of this.

There's an overlay that gets added to all of the slides in the carousel. This example is much darker than it should be. WIthout the overlay, the background would look like this:


We may want to remove the fixed overlay on all slides and only add it in the case of an image or video having that toggle turned on.

## Generic Content Block
I wonder if we should keep the title block here, but have a note that it's not displayed on the frontend? The only reason I mention is because, when collapsed, most of the blocks pull the block title so you can know which block is which. It may be useful to keep the title for cases where multiple Generic Content blocks are used.

**Comment by jomurgel** [comment]
Made a few updates here.

I added the overlay styles to the global partial instead of the hero partial 🤦‍♀️

I also added the overlay toggle function to the video block.

screen shot 2019-03-01 at 10 40 26 am

Which benefits from the styles, but also adds the has-overlay class to elements with the video background.

I also noticed a script error that I've resolved. This may or may not have been related to the animation class consolidation.
screen shot 2019-03-01 at 10 44 00 am

At some point the `data-animation="<?php echo esc_attr( _s_get_animation_class() ); ?>"` code was stripped from the carousel slides and the js wasn't updated, or visa versa.

I've removed the default carousel overlay and rely on the `has-overlay` class to do that work.

Made a few code tweaks:
- Realigned some `=` in the carousel.php.
- Renames the cta button to `button-cta` in the php and style parts.
- Re-added the title to the generic content blocks.
- Updated the text domain in the acf.php link function.

**Comment by coreymcollins** [comment]
Just a quick note – that console error should be taken care of in the PR I submitted this morning (#471).

I'll review this again this afternoon!

**Review by coreymcollins — DISMISSED** [review]
I think we're just about here on this one! With the addition of overlays to everything, it looks like there's a z-index issue with the overlay and the block content:


I'm unable to click on the button here and the text is sitting behind the overlay. This looks to be the case in most of the blocks, though the Related & Recent Posts and Hero block seem fine on my local.

**Comment by jomurgel** [comment]
Hmm 🤔 I thought I'd fixed that.

I've played around and addressed that issue. Looks like I did fix it, but didn't ;) That should be gtg now.

screen shot 2019-03-01 at 1 08 03 pm

I added an acf-block-specific z-index to the `.container` class to ensure all of the content on all of the blocks is above the overlay at all times. Which also adds a relative position on the `.container` class as well.
- https://github.com/WebDevStudios/wd_s/pull/474/commits/47cc66f292063ffb6dd68d36ba90a6388b8b9d48

**Comment by jomurgel** [comment]
@coreymcollins refactored the `block-accordion.php` slightly to remove the wrapper and two internal `.container` divs to instead just have the singular `.container` wrapper. Which also resolves an overlay + background + animation issue.
- https://github.com/WebDevStudios/wd_s/pull/474/commits/ae54ba67aec88c11ddd71601f6853e2af61c329d

Tested the on the other blocks with the same setup, and as far as I can tell we're in good shape.

The only caveat in behavior is the carousel which, because of the touch-swipe feature, the text does not behave as selectable, but the z-index/overlay is correct and the button is pushable. I can't 100% determine if this is expected behavior or some unintended consequence of some style, js, or slick.

**Comment by jomurgel** [comment]
Going to make a few additional changes in reference to #471 to move the font/animation fields to the slides. Will add an additional update when this is complete.

**Comment by jomurgel** [comment]
I've made a couple of updates for #471 in this branch. The changes made in 471 already exists in this branch after pulling in that branch with no changes 👍 

A few things were done here:
- The class name for animation was removed from the parent carousel block (not used).
- I removed font color and animation from the parent and moved those fields to the child.
- Updated the field name for block width in the parent block and applied.
screen shot 2019-03-01 at 2 24 53 pm
- Updated the color class and added to the children.
screen shot 2019-03-01 at 2 25 00 pm
- Updated the [_s_get_animation_class()](https://github.com/WebDevStudios/wd_s/commit/82e081d236e475ae06c45cee52f0c0cfbac7c027) class which provides parsed args for the options as a param and then checks the value for either a string or an array and outputs if there's value.

Screencap of updated anmations and font color.
https://www.dropbox.com/s/c21im5fb69fel75/Kapture%202019-03-01%20at%2014.23.26.gif?dl=0

**Comment by coreymcollins** [comment]
I pushed a handful of additional updates here to take care of a few lingering things I came across:

- Replacing an instance of `wp_kses_post` with our `get_content` function
- Adjusted some z-indexing issues between carousel pagination and the video toggle 
- Replaced some instances of old field names where `hero_slides` was used instead of `carousel_slides`
- Changed the name of the background color field in the Background Options group to `background_color` instead of `color` to stay consistent with `background_image` and `background_video`
- Fixed the output of the block names in the backend

Going to pass this to Will to review as well since it encompasses a fair amount of work I've done now in addition to Jo's.

**Review by ImBigWill — APPROVED** [review]
I think this looks good and ready for merging.

**Comment by coreymcollins** [comment]
This was merged in manually due to the various conflicts popping up. Going to close out!

**Pull request closed without merging**